### PR TITLE
Fix benchmark script

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

### CODE REVIEW

File: benchmark/benchmark.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def summarize(run_dir, metrics, expand_metrics=False):
         # post-processing of report: show a few selected/important metric
         for metric in metrics:
             keys = metric.split(".")
-            value = report
+            value = report.to_dict()
```
Comment (ydshieh) [inline benchmark/benchmark.py:104]: v0.2.0 doesn't need `to_dict`, but `0.4.0` needs this.

Comment (amyeroberts) [reply benchmark/benchmark.py:104]: v0.2.0 of what package?

Comment (ydshieh) [reply benchmark/benchmark.py:104]: sorry, it's `optimum-benchmark`

**Review by amyeroberts — COMMENTED** [review]
Thanks for fixing @ydshieh! 

Change looks OK - but I'm guess it's not backwards compatible if the user has v0.2.0? Is optimum-benchmark version pinned anywhere in the library?

File: benchmark/benchmark.py (hunk 1 above)
Comment (IlyasMoutawwakil) [reply benchmark/benchmark.py:104]: The issue is specifically from this `__post_init__` in v0.2.0 that was incomplete ; only casts the config dict to the appropriate dataclass:
https://github.com/huggingface/optimum-benchmark/blob/v0.2.0/optimum_benchmark/base.py#L25-L29
which was fixed/completed in:
https://github.com/huggingface/optimum-benchmark/blob/v0.3.0/optimum_benchmark/benchmark/base.py#L24-L33

**Comment by ydshieh** [comment]
Thanks. I update the min. version.

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

**Review by ArthurZucker — APPROVED** [review]

**Pull request merged**